weaviate/connect/v4.py

```diff
@@ -144,15 +146,11 @@ def __init__(
         # if there are API keys included add them right away to headers
         if auth_client_secret is not None and isinstance(auth_client_secret, AuthApiKey):
             self._headers["authorization"] = "Bearer " + auth_client_secret.api_key
+            # keeping for backwards compatibility for older clusters for now. On newer clusters, Embedding Service reuses Authorization header.
+            self._headers["X-Weaviate-Api-Key"] = "Bearer " + auth_client_secret.api_key
 
         self._prepare_grpc_headers()
 
-    def __add_weaviate_embedding_service_header(self, wcd_host: str) -> None:
```

Comment:
I would prefer if this function stays, eg if we only add the additonal headers if we are connecting to WCD